File: src/SkyBlock/enchants/bow/BaseBowEnchant.php
<?php


namespace SkyBlock\enchants\bow;


use pocketmine\player\Player;
use SkyBlock\enchants\BaseEnchantment;

class BaseBowEnchant extends BaseEnchantment {
    public function isApplicableTo(Player $holder) : bool {
        return mt_rand(1, 5) === 1;
    }
}

File: SkyblockCore/src/main/java/dev/skyblock/enchants/BaseBowEnchant.java
package dev.skyblock.enchants;
import org.bukkit.entity.Player;
import org.bukkit.event.entity.EntityDamageByEntityEvent;
public abstract class BaseBowEnchant extends BaseEnchant {
    public BaseBowEnchant(int id, String name, int maxLevel) { super(id, name, maxLevel); }
    public abstract void onHitPlayer(Player shooter, Player hit, EntityDamageByEntityEvent ev, int level);
    public boolean isApplicableTo(Player shooter, int level) { return true; }
}
